engine/src/main/java/org/hibernate/search/engine/search/projection/dsl/CommonHighlightProjectionOptionsStep.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.search.projection.dsl;
+
+import java.util.function.Function;
+
+/**
+ * The step in a highlight definition, where optional parameters can be set.
+ */
+public interface CommonHighlightProjectionOptionsStep<N extends ProjectionFinalStep<?>> {
```

Comment:
Not sure introducing this is a good idea... the name is so-so, and it's not a big deal if people always need to call `highlighter(...)` _before_ `.single()`.

Reply:
ok 😃 , maybe like this then? 
the thing is that we'd need to have a `.single()` both after calling highlighter and without calling it  🙈

Reply:
I don't get why you don't just add the `single()` method in `HighlightProjectionOptionsStep` and make it return `ProjectionFinalStep`?

A bit like `FieldProjectionValueStep`.

Anyway... what you did will work fine as well.

Reply:
it's just then `.highlighter(..)` and `single()`  will be in the same interface, and single -- must end up returning `ProjectionFinalStep<String>` while `.highlighter(.. )` can still end with `ProjectionFinalStep<List<String>>` 

and if `.highlighter(..)`  returns the `HighlightProjectionOptionsStep` step, it could work; but users would be able to call   `.highlighter(..)`  multiple times, which doesn't make much sense 🙈

Reply:
> users would be able to call .highlighter(..) multiple times, which doesn't make much sense 🙈

I agree it's not perfect, but it's fine by me. Pretty much every single other setter in `OptionsStep` interfaces can be called multiple times, whether it makes sense or not: it's just simpler that way.

Reply:
ok 👍🏻 😃